compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -505,3 +504,15 @@ showPackageFlag unitId exposeImplicit mods = concat
   where showRenaming (a, b)
           | a == b = GHC.moduleNameString a
           | otherwise = GHC.moduleNameString a <> " as " <> GHC.moduleNameString b
+
+-- NOTE (MK) We used to call just errorIO here. However for reasons
+-- that I do not understand this sometimes seemed to result in test failures
+-- on Windows (never saw it anywhere else) where the executable failed
+-- as expected but we got no output.
+-- So now we are extra careful to make sure that the error message is actually
+-- written somewhere.
+exitWithError :: String -> IO ()
+exitWithError msg = do
+    hPutStrLn stderr msg
+    hFlush stderr
+    exitFailure
```

Comment:
Feels weird to me to exit from anywhere other than a `main`, but I'll leave it to you to decide the style of the project. Is it worth trying to catch higher up and explicitly print and exit there?

Reply:
`exitFailure` still throws an exception if you really want to catch it somewhere up. But generally, compilers can be much more lenient wtr to crashing than long-running processes and if we cannot initialize the package db everything is broken anyway.